projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -119,12 +113,14 @@ internal static class BodySegment
              * +--------------+ */
             public const int FrameSize = BaseFrameSize;
 
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static int WriteTo(Span<byte> span, ushort channel, ReadOnlySpan<byte> body)
             {
                 const int StartBodyArgument = StartPayload;
-
+                span[StartPayload + body.Length] = Constants.FrameEnd;
```

Comment:
If this is before the body.CopyTo, then we  could potentially overwrite it. I'd suggest moving it after, so also to be in line with the other WriteTo methods.

Reply:
Yeah. This was an attempt to reduce bounds checks but should be unnecessary. I'll move it back.